mlir/tools/mlir-tblgen/OpPythonBindingGen.cpp

```diff
@@ -1009,7 +1016,8 @@ static void emitValueBuilder(const Operator &op,
         return (lhs + "=" + llvm::convertToSnakeFromCamelCase(lhs)).str();
       });
   os << llvm::formatv(
-      valueBuilderTemplate,
+      op.getNumResults() > 0 ? valueBuilderTemplate
+                             : valueBuilderNoResultsTemplate,
```

Comment:
should the `if len(op.results) > 0` case be removed from `_get_op_result_or_op_results` with this? it could have a slightly more specific return type